compiler+runtime/src/cpp/jank/runtime/obj/persistent_list.cpp

```diff
@@ -137,7 +137,7 @@ namespace jank::runtime::obj
     {
       return {};
     }
-    return make_box<persistent_list_sequence>(this, ++data.begin(), data.end(), data.size() - 1);
+    return make_box<persistent_list>(data.rest());
   }
 
   persistent_list_sequence_ref persistent_list::next_in_place() const
```

Comment:
The comment in here no longer applies. We should be able to fill this in with a more efficient implementation.